### PR TITLE
These updates add better change detection on configuration attributes for automatic compile iterations.

### PR DESCRIPTION
**Pull request by jmanero-r7**
This includes new code in Rash::merge! to detect updates to the underlying attribute storage, and recursive support for the `dirty` state in namespaces and collections, which was not previously being surfaced to the root attribute set.

### CODE REVIEW

File: lib/builderator/config.rb [hunk 1]
```diff
@@ -47,10 +47,13 @@ def prepend_json(path)
         layers.unshift(File.from_json(path)) if ::File.exist?(path)
       end
 
-      def compile(max_iterations = 4)
+      def compile(max_iterations = 6)
```
Comment (dgreene-r7) [inline lib/builderator/config.rb:50]: Any particular motivation for increasing this? I'm not objecting, it's just odd to me to see a change in a magic number without explanation.

Comment (jmanero-r7) [inline lib/builderator/config.rb:50]: Fixing the change detection code added more iterations

Comment (dgreene-r7) [inline lib/builderator/config.rb:50]: :+1:

**Comment by dgreene-r7** [comment]
LGTM

**Pull request merged**